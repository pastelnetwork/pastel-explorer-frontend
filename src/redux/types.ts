import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { RootState } from '@redux/store';

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export type AppThunkDispatch = ThunkDispatch<RootState, unknown, Action<string>>;
