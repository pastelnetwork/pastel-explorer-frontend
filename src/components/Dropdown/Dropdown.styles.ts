import styled from 'styled-components';

export const Wrapper = styled.div`
  display: inline-flex;
  align-items: center;

  .MuiSelect-icon {
    top: 46%;
    right: 4px;
    transform: translateY(-50%);
  }

  &.hidden-fieldset {
    fieldset {
      display: none;
    }
  }

  .MuiSelect-select {
    width: 150px;
    padding: 6px 28px 7px 10px !important;
    border: 1px solid ${props => props.theme.dropdown.border};
    color: ${props => props.theme.dropdown.color};
    border-radius: 4px;

    &:focus {
      background: transparent;
    }
  }

  .MuiInput-underline {
    &:after,
    &:before {
      display: none;
    }

    &:hover {
      &:not(.Mui-disabled) {
        &:before {
          border-bottom-width: 1px;
        }
      }
    }
  }

  svg {
    color: ${props => props.theme.dropdown.color};
  }
`;

export const Label = styled.div`
  margin-right: 10px;
  color: ${props => props.theme.dropdown.color};
  font-weight: 500;
  white-space: nowrap;
`;
