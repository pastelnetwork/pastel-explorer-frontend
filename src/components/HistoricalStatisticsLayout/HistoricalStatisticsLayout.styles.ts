import styled from 'styled-components/macro';

export const BackButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 12px;
  padding: 14px 16px 14px 0;
  background: ${props => props.theme.card.titleColor};
  border-radius: 10px;
  overflow: hidden;

  ${props => props.theme.breakpoints.down(500)} {
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
  }
`;

export const DropdownWrapper = styled.div`
  ${props => props.theme.breakpoints.down(500)} {
    display: flex;
    justify-content: center;
    width: 100%;
    margin-top: 5px;
  }

  ${props => props.theme.breakpoints.down(380)} {
    .historical-statistics {
      flex-direction: column;
    }
  }
`;

export const BackButton = styled.button`
  display: flex;
  align-items: center;
  border: 0;
  cursor: pointer;
  background: transparent;
  color: ${props => props.theme.palette.text.primary};
  font-size: 16px;
  font-weight: 400;
  line-height: 1.3;
  transition: all 0.5s ease;

  svg {
    transition: background 0.5s ease;
    border-radius: 100%;
  }

  &:hover {
    background: transparent;
    color: ${props => props.theme.sidebar.menu.active};

    svg {
      background: rgba(0, 0, 0, 0.04);
    }
  }
`;

export const ChartWrapper = styled.div`
  border-radius: 10px;
  overflow: hidden;
  box-shadow: 0px 5px 6px rgb(16 16 16 / 6%);
`;
