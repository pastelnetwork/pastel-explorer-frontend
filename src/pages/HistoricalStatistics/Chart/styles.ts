import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
