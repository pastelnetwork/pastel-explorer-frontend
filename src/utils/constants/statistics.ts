import * as routes from '@utils/constants/routes';
import { PeriodTypes, TGranularity } from '@utils/helpers/statisticsLib';
import { TStatisticsInfo } from '@utils/types/IStatistics';
import { TCsvHeaderType } from './types';

export const statistics = [
  {
    id: 'blockchainsize',
    title: 'Blockchain Size',
    url: routes.STATISTICS_BLOCKCHAIN_SIZE,
    image: '/images/statistics/blockchainsize.png',
  },
  {
    id: 'averageBlockSize',
    title: 'Average Block Size',
    url: routes.STATISTICS_AVERAGE_BLOCK_SIZE,
    image: '/images/statistics/averageblocksize.png',
  },
  {
    id: 'difficulty',
    title: 'Difficulty',
    url: routes.STATISTICS_DIFFICULTY,
    image: '/images/statistics/difficulty.png',
  },
  {
    id: 'pslPrice',
    title: 'Price',
    url: routes.STATISTICS_PSLPRICE,
    image: '/images/statistics/pslprice.png',
  },
  {
    id: 'hashrate',
    title: 'Hashrate',
    url: routes.STATISTICS_HASHRATE,
    image: '/images/statistics/hashrate.jpg',
  },
  {
    id: 'mempoolsize',
    title: 'Mempool Size',
    url: routes.STATISTICS_MEMPOOL_SIZE,
    image: '/images/statistics/mempoolsize.jpg',
  },
  {
    id: 'nettotals',
    title: 'Network Total',
    url: routes.STATISTICS_NETTOTALS,
    image: '/images/statistics/nettotals.jpg',
  },
  {
    id: 'transactionsCount',
    title: 'Transaction Count',
    url: routes.STATISTICS_TRANSACTION_COUNT,
    image: '/images/statistics/transactionscount.png',
  },
  {
    id: 'totaltransactioncount',
    title: 'Total Transaction Count',
    url: routes.STATISTICS_TOTAL_TRANSACTION_COUNT,
    image: '/images/statistics/totaltransactioncount.png',
  },
  {
    id: 'transactionfee',
    title: 'Average Transaction Fee',
    url: routes.STATISTICS_TRANSACTION_FEE,
    image: '/images/statistics/transactionfee.jpg',
  },
  {
    id: 'transactionPerSecond',
    title: 'Transaction Per Second',
    url: routes.STATISTICS_TRANSACTION_PER_SECOND,
    image: '/images/statistics/transactionspersecond.jpg',
  },
  {
    id: 'transactionInBlock',
    title: 'Transactions In Block',
    url: routes.STATISTICS_TRANSACTION_IN_BLOCK,
    image: '/images/statistics/transactionsinblock.jpg',
  },
  {
    id: 'averageTransactionsPerBlock',
    title: 'Average Transactions Per Block',
    url: routes.STATISTICS_AVERAGE_TRANSACTIONS_PER_BLOCK,
    image: '/images/statistics/averagetransactionsperblock.png',
  },
];

export const themes = [
  {
    name: 'theme1',
    backgroundColor: '#0d0d0d',
    splitLineColor: '#202021',
    color: '#abaac1',
  },
  {
    name: 'theme2',
    backgroundColor: '#FFF7C6',
    stack: 'confidence-band',
    splitLineColor: '#C7C4CC',
    smooth: true,
    color: '#100c2a',
  },
  {
    name: 'theme3',
    backgroundColor: '#FFF',
    splitLineColor: '#EEE',
    color: '#202021',
  },
];

const commonCsvFields = [
  { label: 'Value', key: 'value' },
  { label: 'Created time', key: 'time' },
];

export const csvHeaders: TCsvHeaderType = {
  averageblocksize: commonCsvFields,
  difficulty: commonCsvFields,
  hashrate: commonCsvFields,
  mempoolsize: commonCsvFields,
  networktotals: [
    { label: 'Receive, Sent', key: 'value' },
    { label: 'Created time', key: 'time' },
  ],
  transactionfee: [
    { label: 'Transaction Fee', key: 'value' },
    { label: 'Created time', key: 'time' },
  ],
  transactionsinblock: [
    { label: 'Block Height', key: 'height' },
    { label: 'Transactions', key: 'transactions' },
    { label: 'Created time', key: 'time' },
  ],
  transactionspersecond: [
    { label: 'Transactions', key: 'value' },
    { label: 'Created time', key: 'time' },
  ],
};

export const pricesCSVHeaders = [
  { label: 'USD Price', key: 'usd' },
  { label: 'BTC Price', key: 'btc' },
  { label: 'Created time', key: 'time' },
];

export const periods: PeriodTypes[][] = [
  ['2h', '2d', '4d', 'all'],
  ['30d', '60d', '180d', '1y', 'all'],
  ['1h', '3h', '6h', '12h'],
];

export const CHART_THEME_BACKGROUND_DEFAULT_COLOR = '#0d0d0d';
export const CHART_THEME_BACKGROUND_DEFAULT_COLOR_LIGHT = '#fff';
export const BLOCK_CHART_DEFAULT_GRANULARITY = '1d';

export const CHART_DEFAULT_PERIOD = 'all';

export const info: TStatisticsInfo = {
  connections: 8,
  currencyName: 'PSL',
  disconnected: false,
  latestBlock: 71976,
  pslPrice: undefined,
  solps: 2652525,
  testnet: false,
  verificationProgress: 0.9999843360337557,
  version: 1000029,
};

export const granularities: TGranularity[][] = [['1d', '30d', '1y']];
