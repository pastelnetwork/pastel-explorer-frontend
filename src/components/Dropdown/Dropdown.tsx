import * as React from 'react';

import MenuItem from '@mui/material/MenuItem';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import Checkbox from '@mui/material/Checkbox';
import ListItemText from '@mui/material/ListItemText';
import Input from '@mui/material/Input';

import { translateDropdown } from '@utils/helpers/i18n';

import * as Styles from './Dropdown.styles';

export type OptionsProps = {
  value: string | number;
  name: string;
};

type DropdownProps = {
  value: string;
  onChange: (_value: SelectChangeEvent) => void;
  options: OptionsProps[];
  label?: React.ReactNode;
  classNameWrapper?: string;
};

export const Dropdown = ({
  value,
  onChange,
  options,
  label,
  classNameWrapper,
}: DropdownProps): JSX.Element => {
  return (
    <Styles.Wrapper className={classNameWrapper}>
      {label ? <Styles.Label>{label}</Styles.Label> : null}
      <Select
        value={value as string}
        onChange={onChange}
        MenuProps={{
          style: {
            maxHeight: '50vh',
          },
        }}
      >
        {options.map((option: OptionsProps) => (
          <MenuItem value={option.value} key={option.value}>
            {option.name}
          </MenuItem>
        ))}
      </Select>
    </Styles.Wrapper>
  );
};

Dropdown.defaultProps = {
  label: undefined,
  classNameWrapper: '',
};

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 250,
    },
  },
};

interface ITagDropdown {
  defaultValues?: string[];
  onChange?: (_values: string[]) => void;
  placeholder?: string;
  options: OptionsProps[];
}

export const TagDropdown: React.FC<ITagDropdown> = ({
  defaultValues = [],
  onChange,
  placeholder,
  options,
}) => {
  const [values, setValues] = React.useState<string[]>(defaultValues);

  const handleChange = (event: SelectChangeEvent<typeof values>) => {
    setValues(
      typeof event.target.value === 'string'
        ? event.target.value.split(',')
        : (event.target.value as string[]),
    );
  };

  const handleClose = () => {
    if (onChange) {
      onChange(values);
    }
  };

  return (
    <Styles.Wrapper>
      <Select
        multiple
        displayEmpty
        value={values}
        onChange={handleChange}
        onClose={handleClose}
        input={<Input />}
        renderValue={selected => {
          if (!(selected as string[]).length) {
            return placeholder;
          }
          const items = options
            .filter(o => o.value && (selected as string[]).indexOf(o.value.toString()) !== -1)
            .map(o => translateDropdown(o.name));
          return items.join(', ');
        }}
        MenuProps={MenuProps}
      >
        {options.map(option => (
          <MenuItem key={option.value} value={option.value}>
            <Checkbox checked={values.indexOf(option.value.toString()) > -1} />
            <ListItemText primary={translateDropdown(option.name)} />
          </MenuItem>
        ))}
      </Select>
    </Styles.Wrapper>
  );
};

TagDropdown.defaultProps = {
  defaultValues: [],
  placeholder: '',
  onChange: () => {},
};
