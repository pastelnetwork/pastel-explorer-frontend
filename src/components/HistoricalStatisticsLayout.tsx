import { memo, ReactNode } from 'react';
// import { BackIcon } from '@components/Icons';
import { makeStyles } from '@material-ui/styles';
import { useHistory } from 'react-router-dom';
import { TAppTheme } from '@theme/index';

interface IProps {
  children: ReactNode;
  currentBgColor: string;
}

const useStyles = makeStyles((theme: TAppTheme) => ({
  root: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
  },
  button: {
    border: 0,
    outline: 'unset',
    marginLeft: theme.spacing(4),
    marginTop: theme.spacing(4),
    background: 'inherit',
    cursor: 'pointer',
    [theme.breakpoints.down('sm')]: {
      marginLeft: theme.spacing(2),
    },
  },
  icon: {
    fill: theme.palette.text.primary,
    stroke: theme.palette.text.primary,
    width: 24,
    height: 22,
    [theme.breakpoints.down('sm')]: {
      width: 19,
      height: 17,
    },
  },
}));

const HistoricalStatisticsLayout = ({ children, currentBgColor }: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  return (
    <div className={classes.root}>
      <div style={{ flex: 1, backgroundColor: currentBgColor }}>
        <div>
          <button type="button" className={classes.button} onClick={() => history.goBack()}>
            <svg
              width="19"
              height="17"
              className={classes.icon}
              viewBox="0 0 19 17"
              fill="black"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0938 7.71894H2.5082L8.47617 2.03457C8.78867 1.73691 8.80078 1.24238 8.50313 0.929878C8.20586 0.617769 7.71133 0.605269 7.39844 0.902925L0.582812 7.39511C0.287891 7.69042 0.125 8.08261 0.125 8.50019C0.125 8.91738 0.287891 9.30996 0.596484 9.61816L7.39883 16.0971C7.55 16.2412 7.74375 16.3127 7.9375 16.3127C8.14375 16.3127 8.35 16.2314 8.50352 16.0701C8.80117 15.7576 8.78906 15.2635 8.47656 14.9658L2.48359 9.28144H18.0938C18.525 9.28144 18.875 8.93144 18.875 8.50019C18.875 8.06894 18.525 7.71894 18.0938 7.71894Z"
                // fill="black"
                // stroke="black"
                strokeWidth="0.2"
              />
            </svg>
          </button>
        </div>
        {children}
      </div>
    </div>
  );
};

export default memo(HistoricalStatisticsLayout);
