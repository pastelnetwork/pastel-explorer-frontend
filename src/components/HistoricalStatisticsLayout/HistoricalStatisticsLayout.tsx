import { memo, ReactNode, useState } from 'react';
// import { BackIcon } from '@components/Icons';
import { makeStyles } from '@material-ui/styles';
import { useHistory } from 'react-router-dom';
import { Skeleton } from '@material-ui/lab';
import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
import parse from 'html-react-parser';
// application
import { Dropdown, OptionsProps } from '@components/Dropdown/Dropdown';
import { statistics } from '@utils/constants/statistics';
import { translate, translateDropdown } from '@utils/helpers/i18n';
import { getCurrencyName } from '@utils/appInfo';

import * as Styles from './HistoricalStatisticsLayout.styles';

interface IProps {
  children: ReactNode;
  currentBgColor: string;
  title?: string | ReactNode;
}

const useStyles = makeStyles(() => ({
  root: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
  },
}));

const HistoricalStatisticsLayout = ({ children, currentBgColor, title }: IProps) => {
  const classes = useStyles();
  const [selectedChart, setSelectedChart] = useState('0');
  const history = useHistory();

  const handleDropdownChange = (
    event: React.ChangeEvent<{
      name?: string | undefined;
      value: unknown;
    }>,
  ) => {
    if (event.target.value) {
      const url = event.target.value as string;
      setSelectedChart(url);
      if (url !== '0') {
        history.push(url);
      }
    }
  };

  const generateChartOption = () => {
    const results: OptionsProps[] = [
      {
        name: title && title !== '[object Object]' ? title?.toString() : '',
        value: '0',
      },
    ];
    for (let i = 0; i < statistics.length; i += 1) {
      if (statistics[i].title !== title) {
        results.push({
          name: translateDropdown(statistics[i].title, { currency: getCurrencyName() }),
          value: statistics[i].url,
        });
      }
    }
    return results;
  };

  return (
    <div className={classes.root}>
      <div style={{ flex: 1 }}>
        <Styles.BackButtonWrapper>
          <Styles.BackButton type="button" onClick={() => history.goBack()}>
            <NavigateBeforeIcon /> {parse(translate('components.historicalStatisticsLayout.back'))}
          </Styles.BackButton>
          <Styles.DropdownWrapper>
            <Dropdown
              value={selectedChart}
              onChange={handleDropdownChange}
              options={generateChartOption()}
              label={`${translateDropdown(
                'components.historicalStatisticsLayout.historicalStatistic',
              )}:`}
              classNameWrapper="historical-statistics"
            />
          </Styles.DropdownWrapper>
        </Styles.BackButtonWrapper>
        <Styles.ChartWrapper style={{ backgroundColor: currentBgColor }}>
          {children || <Skeleton animation="wave" variant="rect" height={386} />}
        </Styles.ChartWrapper>
      </div>
    </div>
  );
};

HistoricalStatisticsLayout.defaultProps = {
  title: '',
};

export default memo(HistoricalStatisticsLayout);
