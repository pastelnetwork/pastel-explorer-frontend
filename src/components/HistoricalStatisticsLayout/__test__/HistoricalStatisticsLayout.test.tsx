import { shallow } from 'enzyme';
import { Skeleton } from '@material-ui/lab';
import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import { Dropdown } from '../../Dropdown/Dropdown';
import i18next from '../../../utils/helpers/i18n';
import { CHART_THEME_BACKGROUND_DEFAULT_COLOR } from '../../../utils/constants/statistics';
import HistoricalStatisticsLayout from '../HistoricalStatisticsLayout';
import * as Styles from '../HistoricalStatisticsLayout.styles';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;

describe('components/HistoricalStatisticsLayout', () => {
  const wrapper = shallow(
    <HistoricalStatisticsLayout currentBgColor={CHART_THEME_BACKGROUND_DEFAULT_COLOR}>
      <div>Content</div>
    </HistoricalStatisticsLayout>,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <NavigateBeforeIcon>', () => {
    expect(wrapper.find(NavigateBeforeIcon).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Dropdown>', () => {
    expect(wrapper.find(Dropdown).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.BackButtonWrapper>', () => {
    expect(wrapper.find(Styles.BackButtonWrapper).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.BackButton>', () => {
    expect(wrapper.find(Styles.BackButton).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.ChartWrapper>', () => {
    expect(wrapper.find(Styles.ChartWrapper).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Skeleton>', () => {
    expect(wrapper.find(Skeleton).length).toBe(0);
  });
});
