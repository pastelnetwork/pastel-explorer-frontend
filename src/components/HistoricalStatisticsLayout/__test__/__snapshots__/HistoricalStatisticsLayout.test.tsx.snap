// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HistoricalStatisticsLayout renders correctly 1`] = `ShallowWrapper {}`;
